shield/internal/evaluator/evaluator.go

```diff
@@ -12,7 +12,11 @@ import (
 func Eval(exp *ast.Expression, env env.Environment) object.Object {
 	switch exp := exp.Value.(type) {
 	case *ast.Expression_IntegerLiteral:
-		return &object.Integer{Value: exp.IntegerLiteral.Value}
+		value, success := new(big.Int).SetString(exp.IntegerLiteral.Value, 10)
+		if !success {
+			return newError("failed to parse integer: " + exp.IntegerLiteral.Value)
```

Comment:
this is not really a parsing error (from the parser) so I would rephrase it a little to avoid confusion:

```suggestion
			return newError("invalid IntegerLiteral value: " + exp.IntegerLiteral.Value)
```